src/Tribe/Main.php

```diff
@@ -3919,17 +3919,113 @@ public function isOrganizer( $postId = null ) {
 		}
 
 		/**
-		 * Get a "previous/next post" link for events. Ordered by start date instead of ID.
+		 * Modify the WHERE clause of query when fetching next/prev posts so events with identical times are not excluded
+		 *
+		 * This method ensures that when viewing single events that occur at a given time, other events
+		 * that occur at the exact same time are are not excluded from the prev/next links
+		 *
+		 * @since 4.0.2
+		 *
+		 * @param string $where_sql WHERE SQL statement
+		 * @param WP_Query $query WP_Query object
+		 *
+		 * return string
+		 */
+		public function get_closest_event_where( $where_sql ) {
+			// if we are in this method, we KNOW there is a section of the SQL that looks like this:
+			//     ( table.meta_key = '_EventStartDate' AND CAST( table.meta_value AS DATETIME ) [<|>] '2015-01-01 00:00:00' )
+			// What we want to do is to extract all the portions of the WHERE BEFORE that section, all the
+			// portions AFTER that section, and then rebuild that section to be flexible enough to include
+			// events that have the SAME datetime as the event we're comparing against.  Sadly, this requires
+			// some regex-fu.
+			//
+			// The end-game is to change the known SQL line (from above) into the following:
+			//
+			//  (
+			//    ( table.meta_key = '_EventStartDate' AND CAST( table.meta_value AS DATETIME ) [<|>] '2015-01-01 00:00:00' )
+			//    OR (
+			//      ( table.meta_key = '_EventStartDate' AND CAST( table.meta_value AS DATETIME ) = '2015-01-01 00:00:00' )
+			//      AND
+			//      table.post_id [<|>] POST_ID
+			//    )
+			//  )
```

Comment:
Bear with me here because I'm only on :coffee: #1, but let's say we start with the following data:

```
ID  EventStartDate
1   2016-01-01 12:00:00
2   2016-02-10 12:00:00
3   2016-03-09 15:00:00
4   2016-03-09 15:00:00
5   2016-03-10 16:00:00
```

If we start with the user landing on event `#2` - this works well. The next event link will take them to event `#3`, then to event `#4` without skipping ahead to event `#5`. All good there - but what if the event data is then edited to the following (see the change in `#1`'s date/time):

```
ID  EventStartDate
1   2016-03-09 15:00:00
2   2016-02-10 12:00:00
3   2016-03-09 15:00:00
4   2016-03-09 15:00:00
5   2016-03-10 16:00:00
```

Again starting with a user visiting event `#2`, the first bit of the amended where clause will be matched (a date time greater than current) and we'll go to event `#3` and then `#4`, `#5` ... but we skip `#1`. 

Does that make sense and is it a concern for us to address here? Or am I just confused :confounded: